lib/fetchers/TokenFetcher.ts

```diff
@@ -58,12 +58,26 @@ export class TokenFetcher {
     }
   };
 
-  public getTokenByAddress = async (chainId: ChainId, address: string): Promise<Currency | undefined> => {
-    if (address == NATIVE_ADDRESS || address.toLowerCase() == 'eth') {
+  /**
+   * Gets the token currency object if the address or symbol is found in the DEFAULT_TOKEN_LIST.
+   * Returns undefined if the token is not found.
+   */
+  public getTokenBySymbolOrAddress = async (
+    chainId: ChainId,
+    symbolOrAddress: string
+  ): Promise<Currency | undefined> => {
+    // check for native symbols first
+    if (NATIVE_NAMES_BY_ID[chainId]!.includes(symbolOrAddress) || symbolOrAddress == NATIVE_ADDRESS) {
       return Ether.onChain(chainId);
     }
 
     const tokenListProvider = this.getTokenListProvider(chainId);
-    return await tokenListProvider.getTokenByAddress(address);
+
+    // try to fetch token by symbol first
+    const token = await tokenListProvider.getTokenBySymbol(symbolOrAddress);
+    if (token !== undefined) return token;
+
+    // if not found, try to fetch token by address
+    return await tokenListProvider.getTokenByAddress(symbolOrAddress);
```

Comment:
Can't we call `resolveTokenBySymbolOrAddress` above and then `tokenListProvider.getTokenByAddress(resolvedAddress);` ? 

That way the hot path (token is an address) we only make one call and if its a symbol we make two

Reply:
yep good idea